### PR TITLE
Fixes for various issues spotted during testing

### PR DESCRIPTION
**Pull request by Graeme-Miller**

### CODE REVIEW

File: common/src/main/resources/docker/docker.bom [hunk 1]
```diff
@@ -433,6 +433,10 @@ brooklyn.catalog:
 
         launch.command: |
           docker run -d ${IMAGE_DETAILS}
+          
+        checkRunning.command: |  
+          result="$(docker ps | grep ${IMAGE_DETAILS} | wc -l)"
+          if [ $result -ne 0 ]; then exit 0; else exit 1; fi;
```
Comment (neykov) [inline common/src/main/resources/docker/docker.bom]: See how `docker-vm-container` implements this, could re-use.

File: common/src/main/resources/docker/docker.bom [hunk 2]
```diff
@@ -509,8 +513,8 @@ brooklyn.catalog:
           }
           rm -f "${PID_FILE}" # docker won't overwrite
           DOCKER="docker run -d --cidfile \"${PID_FILE}\" --net=host"
-          [ "${DOCKER_RUN_VOLUMES}" ] && DOCKER="${DOCKER} $( parse_docker_volumes )"
-          [ "${DOCKER_RUN_ENV}" ] && DOCKER="${DOCKER} $( parse_docker_env )"
+          [[ $DOCKER_RUN_VOLUMES != [] ]] && DOCKER="${DOCKER} $( parse_docker_volumes )"
+          [[ $DOCKER_RUN_ENV != {} ]] && DOCKER="${DOCKER} $( parse_docker_env )"
```
Comment (neykov) [inline common/src/main/resources/docker/docker.bom]: Could any of those still be the empty string in which case the check will be a success?

Comment (Graeme-Miller) [inline common/src/main/resources/docker/docker.bom]: I don't believe so, these are set from a DSL brooklyn:config call.
There are tests which cover the config being empty and the config being full- they both pass with this change.

File: common/src/main/resources/docker/docker.bom (hunk 1 above)
Comment (Graeme-Miller) [inline common/src/main/resources/docker/docker.bom]: good point, but I wanted to check that the specific image was running.
Do you think it is ok to merge without?

**Comment by neykov** [comment]
Can be merged after resolving merge conflicts.

**Pull request merged**